Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -30,26 +28,22 @@ public final class DefaultHostResolver: HostResolver {
         let shutdownCallbackOptionPtr = shutDownOptions?.toShutDownCPointer()
 
         self.shutDownOptions = shutDownOptions
-        let options = aws_host_resolver_default_options(max_entries: maxHosts,
-                                                        el_group: elg.rawValue,
-                                                        shutdown_options: shutdownCallbackOptionPtr,
-                                                        system_clock_override_fn: nil)
-        self.hostResolverOptionsPtr = fromPointer(ptr: options)
-        self.rawValue = aws_host_resolver_new_default(allocator.rawValue, hostResolverOptionsPtr)
-
-        let config = aws_host_resolution_config(
-            impl: aws_default_dns_resolve,
-            max_ttl: maxTTL,
-            impl_data: nil
-        )
-
-        self.config = fromPointer(ptr: config)
 
+        var options = aws_host_resolver_default_options(max_entries: maxHosts,
```

Comment:
trivial: I think the reason the old code was crazy, and the reason you have to use `var` here instead of `let` is just that the C function forgot to mark the options arg as `const`

we should fix the C function. "options" structs should always be passed as `const`